### ISSUE TITLE
Disable validation when object is added as nested_form

### ISSUE THREAD

**Issue by Rubioli** [DavyJonesLocker/client_side_validations#709]
Great gem 👍 

I have models `Post` &  `Image` and my models are set up as:
```
class Post < ActiveRecord::Base
  validates :title, :price, presence: true
  validates :price, numericality: {greater_than_or_equal_to: 0.010, message: "Can't be less then $0.010" }
  has_many :images, inverse_of: :post
  accepts_nested_attributes_for :images, reject_if: :all_blank, allow_destroy: true

class Image < ActiveRecord::Base
  validates :title, :price, presence: true
  validates :price, numericality: {greater_than_or_equal_to: 0.010, message: "Can't be less then $0.010" }
  belongs_to :post
```

In my application user can add many images under a post & I'm have my nested_forms inside `post#_image_fields.html.haml`.

*My `posts#new` is set up as:*
```
= simple_form_for @post, validate: true do |f|
  .div
    // Post form inputs
    = f.input :title, label: 'Campaign Title'
    = f.input :price, label: 'CPC - Rest of World'

  #images
    = f.simple_fields_for :images, validate: false do |image|
      = render 'image_fields', f: image
```

*My `posts#image_fields` is set up as:*
```
.nested-fields
  = f.text_field :title, validate: false
  = f.text_field :description
```
*PS: As you can see I don't have `:price` in my `posts#image_fields` for images & its because image will inherit post's `price`and its done from model with a `after_create`.*

So Basically what I'm trying to do here is to make `images` **ONLY** be validated when its in `images#edit` and not when its submitted as `nested_forms` in `posts#image_fields`.

I've tried to add `validate: false` to `text_field` & `simple_fields_for` but that doesn't disable the validation on `:title & :price`.

Any help is appreciated :) 

*I am using `cocoon gem` for nested_forms & Rails 4.2.4 & client_side_validations 4.2.3*

**Comment by tagliala** [comment]
Hi,

thanks for reporting this.

CSV issues are hard to debug and extremely time consuming to investigate.

You should provide all the information required by the [Contributing Guidelines](https://github.com/DavyJonesLocker/client_side_validations/blob/master/CONTRIBUTING.md#submitting-a-new-issue)

The best you can do to speed up things is to share a new application with just the minimum amount of code to reproduce the issue (such as: one model, one controller, one form with two file fields), the minimum amount of gems (the basic rails gemfile + csv + csv_simple-form + paperclip + mongoid_paperclip)

Please also make sure to use the latest version of CSV because I do not support CSV for Rails 4

**Comment by Rubioli** [comment]
Thanks for your reply.
I have two questions before starting with setting up the dummy application.

1) Can validation be skipped when creating new object and only apply of update?
2) CSV doesn't support Rails 4 so I can't use it in Rails 4 application?

Appreciate your help :)

**Comment by tagliala** [comment]
> 1. Can validation be skipped when creating new object and only apply on update/edit?

I'm not sure I understand the issue.

Given that CSV is not involved in server side validations (is [this](http://guides.rubyonrails.org/active_record_validations.html#on) involved?), you could use different templates or do something like `validate: object.persisted?` to enable client side validations only when the object is already persisted in the database (= editing)

> 2. CSV doesn't support Rails 4 so I can't use it in Rails 4 application? 

CSV 4.x does support Rails 4, but I'm not going to support CSV 4 anymore. So, if the issue is still present in the current version of CSV, I will try my best to fix it and then you can backport the fix to your own fork of CSV 4.

> 3. What is the latest version CSV and how to update? I installed the gem few weeks ago with gem 'client_side_validations'

CSV 9.3.3 at the moment I'm writing. The gem will automatically update if you use Rails 5.x. Anyway, upgrading from Rails 4 to Rails 5 is not quite simple so I can understand if you can't

Hope it helps

**Comment by Rubioli** [comment]
Hello @tagliala Thanks for your reply.

Here you have the project: https://github.com/Rubioli/csv_test

As you can see if you go to http://localhost:3000/posts/new, you can't create a post and its because `:price` is `present: true` from `Image` model but not added in that `nested_form` and image `:price` will be added/inherit from `Post` model with `after_create :add_price`.

So basically I try to fix is to disable or  turned of **Image validation** when creating new **post (http://localhost:3000/posts/new)** But when editing image (http://localhost:3000/images/1/edit), I want `title` & `price` to be validated.

Appreciate you help :)

**Comment by tagliala** [comment]
I'm going to close here because I don't have the resources to support Rails 4

Hope you solved meanwhile
